src/drivers/intel/dmic/dmic_computed.c

```diff
@@ -132,7 +132,7 @@ static void find_modes(struct dai *dai,
 			mfir = fir_list[j]->decim_factor;
 
 			/* Skip if previous decimation factor was the same */
-			if (j > 1 && fir_list[j - 1]->decim_factor == mfir)
+			if (j != 0 && fir_list[j - 1]->decim_factor == mfir)
```

Comment:
this is an actual flow change, is this intended? @singalsu is this correct? If this is fixing an error, we should indicate that. Also not sure why you're saying that this check is pointless?

Reply:
Yep, there's a mistake, also a duplicate decimation factor in fir_list[1] should be checked for. The check should have been `if (j > 0 &&` but check for non-zero is fine too. The commit title and description could be changed.